sklearn/model_selection/tests/test_split.py

```diff
@@ -1138,6 +1162,11 @@ def test_check_cv():
     cv = check_cv(3, y_multiclass, classifier=True)
     np.testing.assert_equal(list(StratifiedKFold(3).split(X, y_multiclass)),
                             list(cv.split(X, y_multiclass)))
+    # also works with 2d multiclass
+    y_multiclass_2d = y_multiclass.reshape(-1, 1)
+    cv = check_cv(3, y_multiclass_2d, classifier=True)
+    np.testing.assert_equal(list(StratifiedKFold(3).split(X, y_multiclass_2d)),
+                            list(cv.split(X, y_multiclass_2d)))
```

Comment:
it's worth testing that it differs from KFold(3) to be sure that the assertion is meaningful

Reply:
assertion os copied from above. Should I add a check there, too?

Reply:
I suppose so